types/posthtml.d.ts

```diff
@@ -1,24 +1,23 @@
-export interface NodeAttributes {
+type MaybeArray<T> = T | T[]
+
+export interface PostHTMLNodeAttributes {
   [attribute: string]: string;
 }
 
-export type NodeContent = Array<Node | PostHTMLTree | PostHTMLTree[]>;
+export type PostHTMLNodeContent = string | PostHTMLNode;
+
+export type PostHTMLMatcher = string | RegExp | object;
+export type PostHTMLExpression = MaybeArray<PostHTMLMatcher>;
+export type PostHTMLNodeCallback = (node: PostHTMLNode) => MaybeArray<PostHTMLNode>;
 
 // T - Tag name
 // A - Attributes
-export interface Node<T = string, A = NodeAttributes> {
+export interface PostHTMLNode<T = string, A = PostHTMLNodeAttributes> {
```

Comment:
💄 IMHO, `T` and `A` — not a good title for type-parameter that leads to the comment above. We can use `TTag` and `TAttributes` instead.